icu4c/source/common/uloc_tag.cpp

```diff
@@ -858,15 +858,15 @@ _addVariantToList(VariantListEntry **first, VariantListEntry *var) {
             cmp = uprv_compareInvCharsAsAscii(var->variant, cur->variant);
             if (cmp == 0) {
                 /* duplicated variant */
-                bAdded = false;
-                break;
+                return false;
             }
             prev = cur;
             cur = cur->next;
         }
     }
 
-    return bAdded;
+    var.orphan();
```

Comment:
I think it's cleaner if you replace the var.getAlias() with var.orphan() and remove this line here.

Reply:
Done.